src/pages/index.js

```diff
@@ -80,6 +108,58 @@ const previewExitButton = previewModal.querySelector(".modal__exit-button");
 const previewImage = previewModal.querySelector(".modal__image");
 const previewCaption = previewModal.querySelector(".modal__caption");
 
+// avatar elements
+const avatarOpenButton = document.querySelector(".profile__avatar-button");
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarCloseButton = avatarModal.querySelector(".modal__exit-button");
+const avatarForm = document.forms["avatar-form"];
+const avatarFormLink = avatarForm.querySelector("#avatar-input-link");
+
+// delete elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteCloseButton = deleteModal.querySelector(".modal__exit-button");
+const deleteCancelButton = deleteModal.querySelector("#cancel-button");
+const deleteConfirmButton = deleteModal.querySelector("#delete-button");
+
+// delete events
+deleteCloseButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+deleteCancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+deleteConfirmButton.addEventListener("click", () => {
+  deleteConfirmButton.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId)
+    .then((res) => {
+      closeModal(deleteModal);
+      selectedCard.remove();
+    })
+    .catch(console.error)
+    .finally(() => (deleteConfirmButton.textContent = "Delete"));
+});
+
+// avatar events
+avatarOpenButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+avatarCloseButton.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.